From 301600758f18178b5cfc65626c8fb498c0ad2bfa Mon Sep 17 00:00:00 2001
From: Tomohiro Kashiwada <kikairoya@gmail.com>
Date: Tue, 8 Jul 2025 04:55:53 +0900
Subject: [PATCH] [libclang][Cygwin] Use __declspec(dllexport) for libclang on
 Cygwin (#147122)

This is needed for Cygwin build without `-DLLVM_LINK_LLVM_DYLIB=ON`,
otherwise causes a linker error 'export ordinal too large'.
---
 clang/include/clang-c/Platform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clang/include/clang-c/Platform.h b/clang/include/clang-c/Platform.h
index 67c1fff8ff..8d341ddd6f 100644
--- a/clang/include/clang-c/Platform.h
+++ b/clang/include/clang-c/Platform.h
@@ -22,7 +22,7 @@ LLVM_CLANG_C_EXTERN_C_BEGIN
 #ifndef CINDEX_NO_EXPORTS
   #define CINDEX_EXPORTS
 #endif
-#ifdef _WIN32
+#if defined(_WIN32) || defined(__CYGWIN__)
   #ifdef CINDEX_EXPORTS
     #ifdef _CINDEX_LIB_
       #define CINDEX_LINKAGE __declspec(dllexport)
-- 
2.50.1.windows.1

